editor/src/clj/editor/app_view.clj

```diff
@@ -772,16 +772,26 @@
 (defn- decorate-target [engine-descriptor target]
   (assoc target :engine-id (:id engine-descriptor)))
 
-(defn- launch-engine! [engine-descriptor project-directory prefs debug?]
+(defn- launch-engine! [engine-descriptor project-directory prefs debug? workspace]
   (try
     (report-build-launch-progress! "Launching engine...")
     (let [engine (engine/install-engine! project-directory engine-descriptor)
-          launched-target (->> (engine/launch! engine project-directory prefs debug?)
-                               (decorate-target engine-descriptor)
-                               (targets/add-launched-target!)
-                               (targets/select-target! prefs))]
-      (report-build-launch-progress! (format "Launched %s" (targets/target-message-label launched-target)))
-      launched-target)
+          count (prefs/get-prefs prefs (prefs/make-project-specific-key "instance-count" workspace) 1)
+          pause-ms 100
+          instance-index-range (if (= count 1) (range (inc 0)) (range 1 (inc count)))
+          launched-targets (for [instance-index instance-index-range]
+                             (let [last-instance? (or (= count 1) (= instance-index count))
+                                   instance-debug? (and debug? last-instance?)
+                                   launched-target (->> (engine/launch! engine project-directory prefs instance-debug? instance-index)
+                                                        (decorate-target engine-descriptor)
+                                                        (targets/add-launched-target! instance-index))]
+                               (when (not last-instance?)
+                                 (Thread/sleep pause-ms))        ;pause needed to make sure the launch order of instances is right
```

Comment:
If seems like it's not the main UI-thread. But even if it is, it shouldn't be a problem, right?